### PR TITLE
Task details

### PR DESCRIPTION
**Pull request by Larionovv**
Change TaskDetails component. 
Create TaskDetailsModal, EditDescription, EditName, TaskPriority, TaskStatus components. 
Create simple style for modal window
Create test for TaskDetails

### CODE REVIEW

File: src/components/TaskDetails/EditDescription.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export default class EditDescription extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editDescription: props.editDescription,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (opanki) [inline src/components/TaskDetails/EditDescription.jsx]: FYI: https://stackoverflow.com/questions/51980977/why-is-componentwillreceiveprops-deprecated

File: src/components/TaskDetails/TaskDetailsStyle.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
+.MainModalTD{
```
Comment (opanki) [inline src/components/TaskDetails/TaskDetailsStyle.scss]: it's common to use camelCase for naming variables, but kebab-case for css classes 

camelCase
snake_case
kebab-case

FYI: https://medium.freecodecamp.org/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849

File: src/components/TaskDetails/TaskPriority.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export default class TaskPriority extends React.Component {
+  handleChangePriority = (priorityName) => {
+    const { changePriority } = this.props;
+    changePriority(false, false, false, priorityName);
+  };
+
+  render() {
+    const { priority } = this.props;
+    switch (priority) {
+      case 'High':
```
Comment (opanki) [inline src/components/TaskDetails/TaskPriority.jsx]: i would suggest to add classes like `priority-${priority}` and describe different colors for different classes in css

File: src/components/TaskDetails/TaskDetailsStyle.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  color: rgb(103, 102, 106);
+}
+.task-details-container{
+  padding: 10px;
+  min-height: 30px;
+  margin: 10px;
+  background-color: rgba(108, 117, 125, 0.07);
+  border-radius: 3px;
+}
+.task-details-container:hover{
```
Comment (opanki) [inline src/components/TaskDetails/TaskDetailsStyle.scss]: you can use nesting here
```
.task-details-container{
    &:hover {

    }
}
```

File: src/components/TaskDetails/TaskDetails.jsx [hunk 5]
```diff
@@ -1,165 +1,129 @@
...
-        </Modal.Footer>
-      </Modal>
+      <Container>
+        <Container className="task-details-container">
+          {nameContext}
+        </Container>
+        <Container className="task-details-container description-container">
+          {descriptionContext}
+        </Container>
+        <ButtonGroup className="ButtonContainer">
```
Comment (mishalisovskiy) [inline src/components/TaskDetails/TaskDetails.jsx]: Classes are usually not camelcased

**Pull request merged**